gr-audio/lib/alsa/alsa_sink.cc

```diff
@@ -369,25 +369,42 @@ int alsa_sink::work_s32(int noutput_items,
     unsigned int nchan = input_items.size();
     const float** in = (const float**)&input_items[0];
     sample_t* buf = (sample_t*)d_buffer;
-    int bi;
+
+    /*
+     * vector of vectors to keep each channel after conversion
+     * to int32_t. These are then placed in the buffer in the
+     * standard WAV file format:
+     * [chan0samp0, chan1samp0, ..., chanNsamp0, chan0samp1, ...]
+     * */
+    volk::vector<volk::vector<sample_t>> buf_channels(
```

Comment:
This should probably be allocated outside of work()

Reply:
I'll put that in the constructor / initialization. Thanks

Reply:
@mbr0wn I was looking into moving this to the constructor but I'm not sure about how to do it given that the output of the work functions are either `int32_t` or `int16_t`. I don't see anywhere in the constructor that knows the output type at initialization (it's determined in `check_topology()`). Do you have any ideas for this? Hope I'm missing something.

Reply:
Nothing wrong with allocating in check_topology(). I think that gets called more than once potentially, so you'll need to check if it's allocated already there.